src/testing/test_runner.cc

```diff
@@ -90,8 +96,11 @@ void test_runner::start_thread(int ac, char** av) {
                 try {
                     auto func = _task.take();
                     return func();
-                } catch (const stop_execution&) {
+                } catch (const stop_execution& se) {
                     _done = true;
+                    if (se.exit_code) {
```

Comment:
> We still return 0 on the present path on success as no seastar tasks should be left behind in this case.

That's not true. The segfaults we have seen so far weren't happening in failing tests, but in passing tests. There were no indication in the test logs that the tests have failed.

I'm not sure what are the tasks which are left behind, but I guess they might be some internal tasks.
For example, in the old case I mentioned in (https://github.com/scylladb/seastar/issues/1514#issuecomment-1438620388), I see:
```
seastar::smp_message_queue::async_work_item<seastar::reactor::do_run()::$_42::operator()(std::unique_ptr<seastar::network_stack, std::default_delete<seastar::network_stack> >) const::{lambda()#1}>::~async_work_item() (this=0x6020000c6dc0) at ./seastar/include/seastar/core/smp.hh:211
```
in the backtrace, even though the test has nothing to do with networking.

Reply:
We don't have good stack traces of other instances of the problem due to the libseastar.so problem, so we don't have a good idea what are the tasks being destroyed in those cases, but perhaps all the instance of this bug are happening to this network_stack lambda?

The only piece of code which matches the signature seems to be:
```
https://github.com/scylladb/seastar/blob/e2785e4b4441ba4f66ef93d0d32f1dc805d6470b/src/core/reactor.cc#L3087-L3092
```

But this task should complete before the test starts (because app::run() waits for it), so I'm not sure what's going on.